UpdatedImageProcessing/TargetDetection/single_targets_capturer.py

```diff
@@ -0,0 +1,116 @@
+           cropped image are: background_color_1, background_color_2, and
+           target_color with background_color_1 having more frequent appearance.
+           The letter_color is either blended with the target_color or shares
+           its color with one of the background_colors.
+        3. Find the frequencies of the three colors' appearance on the image.
+           If neither background_color_1 nor background_color_2 has the most
+           frequent appearance, then the shape_color is set to the color with
+           the most frequent appearance.
+
+           If background_color_1 has the most frequent appearance and
+           background_color_2 has the second most frequent appearance, then
+           shape_color is set to the color with the least frequent appearance.
+
+           If neither of the two cases above applies, then shape_color is set to
+           the color with the second frequent appearance.
+        4. With the knowledge of the shape_color, nullify the background_colors,
+           namely, all the colors that lie outside of the shape's boundary.
+        5. Recrop the image down to the shape's boundary with 5-pixel margins on
+           each side.
+        6. Append the resultant_image into the resultant_image_list.
+        7. Return the resultant_image_list along with list_to_eliminate.
+        """
+        target_map = Image.open(target_map_path)
+        target_map_average_color = BlobColorOperations.find_blob_average_color(target_map, (0, 0, target_map.width, target_map.height))
+
+        resultant_image_list = []
+        list_to_eliminate = []
+        for index in range(len(positive_list)):
+
+            blob_location = positive_list[index]
+
+            blob_x = blob_location[0]
+            blob_y = blob_location[1]
+            blob_width = blob_location[2]
+            blob_height = blob_location[3]
+
+            crop_x1 = blob_x - 20
+            crop_y1 = blob_y - 20
+            crop_x2 = blob_x + blob_width + 20
+            crop_y2 = blob_y + blob_height + 20
+
+            if (crop_x1 < 0):
+                crop_x1 = 0
+
+            if (crop_x2 >= target_map.width):
+                crop_x2 = target_map.width - 1
+
+            if (crop_y1 < 0):
+                crop_y1 = 0
+
+            if (crop_y2 >= target_map.height):
+                crop_y2 = target_map.height - 1
+
+            captured_image = target_map.crop((crop_x1, crop_y1, crop_x2, crop_y2))
+            quantized_image = ColorOperations.quantize_color(captured_image, 4)
+            quantized_image_average_corner_color = BlobColorOperations.find_average_corner_color(quantized_image)
+            percentage_difference = ColorOperations.find_percentage_difference(target_map_average_color, quantized_image_average_corner_color)
+            """
+            if (percentage_difference >= 10):
+
+                list_to_eliminate.append(index)
+                continue
+            """
```

Comment:
Should this code be removed?

Reply:
If these several lines are functional, the code would return less false positives and less correct targets, I was hesitating whether to take it off or not.

Reply:
Can you add a comment to this specifying that this will reduce the number of targets returned?

Reply:
I added the requested comment. Not sure why this change request is still on.